docs/proposals/Ashish_Mahajan_Versioning_Proposal.md

```diff
@@ -0,0 +1,241 @@
+# Proposal : Branch-Based Version Management System for DISCOVER Cookbook
+
+## Abstract
+
+This proposal outlines a comprehensive implementation strategy for a branch-based version management system for the DISCOVER Cookbook following industry-standard documentation versioning practices as used by scikit-learn, Matplotlib, NetworkX, and other NumFOCUS projects. The implementation will enable users to seamlessly navigate between different cookbook editions (v2.0/main and v1.0/master) through a version selector integrated with the existing Jupyter Book interface, while maintaining the current GitHub Pages infrastructure.
+
+## Detailed description
+
+### Current Architecture Analysis
+
+The DISCOVER Cookbook currently maintains two separate versions with no navigation between them:
+
+- Current version (v2.0) in the main branch, deployed at https://discover-cookbook.numfocus.org
+- Previous version (v1.0) in the master branch, deployed at https://discover-cookbook.github.io/
+
+This structure creates several critical challenges:
+
+1. **Hidden Versions**: Users can't tell which version they're viewing or that other versions exist, so they might miss updated recommendations.
+
+2. **Hard to Navigate**: Switching between versions requires typing different URLs or navigating through GitHub, making for a poor user experience.
+
+3. **Content Evolution Opacity**: Users can't easily see how DEI recommendations have changed over time.
+
+
+### Solution Architecture
+
+My implementation follows the industry-standard branch-based versioning model used by scikit-learn, Matplotlib, NetworkX, and other NumFOCUS projects:
+
+1. **Branch-Version Mapping**:
```

Comment:
It's not the best design to make a branch for each of these. I prefer tags that will have tooling around the tag. 

For example you have pandas with only one active branch: https://github.com/pandas-dev/pandas/branches

but the repository has tools that allow it to present other branches: https://pandas.pydata.org/docs/